src/driver/multistage.hpp

```diff
@@ -25,33 +26,37 @@
 
 namespace parthenon {
 
-struct Integrator {
-  Integrator() = default;
-  Integrator(int nstages, std::vector<Real> beta) : nstages(nstages), beta(beta) {}
+struct StagedIntegrator {
+  StagedIntegrator() = default;
+  StagedIntegrator(int nstages, std::vector<Real> beta) : nstages(nstages), beta(beta) {}
+  explicit StagedIntegrator(ParameterInput *pin);
   int nstages;
   std::vector<Real> beta;
+  std::vector<std::string> stage_name;
   Real dt;
 };
 
 class MultiStageDriver : public EvolutionDriver {
  public:
-  MultiStageDriver(ParameterInput *pin, ApplicationInput *app_in, Mesh *pm);
-  std::vector<std::string> stage_name;
-  Integrator *integrator;
-  ~MultiStageDriver() { delete integrator; }
+  MultiStageDriver(ParameterInput *pin, ApplicationInput *app_in, Mesh *pm)
+      : EvolutionDriver(pin, app_in, pm),
+        integrator(std::make_unique<StagedIntegrator>(pin)) {}
+  // An application driver that derives from this class must define this
+  // function, which defines the application specific list of tasks and
+  // there dependencies that must be executed.
+  virtual TaskCollection MakeTaskCollection(BlockList_t &blocks, int stage) = 0;
+  virtual TaskListStatus Step();
 
- private:
+ protected:
+  std::unique_ptr<StagedIntegrator> integrator;
```

Comment:
Question: Why not hold the `StagedIntegrator` directly instead of by `unique_ptr`?  Is there an overarching design reason not to?

Reply:
not a great one.  sometimes it might make sense to pass the integrator object into a task, as is done in the particles example.  since task arguments are captured by value, this was an attempt to discourage copying the object.  Worst case is you copy the `unique_ptr`, but preferably you just pass in the pointer, as is also done in the particles example.

I don't have strong feelings about this at all.  It would also work to just store the `StagedIntegrator` object directly and take it's address when passing it into a task.  Is that preferable?

Reply:
You cannot copy a `unique_ptr` so you you cannot pass it directly by value.  It would have to be passed by pointer or preferably (IMO) by `StagedIntegrator &` or `StagedIntegrator const&`.  I generally tend to prefer passing by reference when the object is guaranteed/required to exist, as is the case here.  My thinking is that one should not have a `MultiStageDriver` without a valid integrator.  The `unique_ptr` should not be `nullptr`, right?

If I were writing the code, I would store it directly here and prefer passing by reference (not pointer) to the task if needed.  

Less important here as the driver is unlikely to be copied (I think), but generally relevant, is that by *not* providing copy ctor/asgn this class also becomes uncopyable because its member is not copyable.  

Just food for thought.  I'm not holding the PR up over this.